### PR TITLE
Misc devs on polynomial chaos expansion

### PR DESCRIPTION
**Pull request by mbaudin47**
- Create new unit tests of CleaningStrategy and FixedStrategy (this was a part of #2330 which is not directly related to PCE cross validation.)
- Renamed LinearModelAnalysis-Identifiers into LinearModelAnalysis-IdentifiersNumber (a part of #2330)
- Update Python examples for new `MetamodelValidation` API.
- Update Python unit test for new `isLeastSquares()` and `involvesModelSelection()` methods.

These commits were originally in #2330.

### CODE REVIEW

File: lib/src/Uncertainty/Algorithm/MetaModel/LinearModel/LinearModelAnalysis.cxx [hunk 1]
```diff
@@ -387,7 +387,7 @@ Graph LinearModelAnalysis::drawModelVsFitted() const
   Cloud cloud(dataFull, "black", "fcircle");
   graph.add(cloud);
   // Add point identifiers for worst residuals
-  UnsignedInteger identifiers(ResourceMap::GetAsUnsignedInteger("LinearModelAnalysis-Identifiers"));
+  UnsignedInteger identifiers(ResourceMap::GetAsUnsignedInteger("LinearModelAnalysis-IdentifiersNumber"));
```
Comment (jschueller) [inline lib/src/Uncertainty/Algorithm/MetaModel/LinearModel/LinearModelAnalysis.cxx:390]: const UnsignedInteger identifiers = ...

Comment (mbaudin47) [reply lib/src/Uncertainty/Algorithm/MetaModel/LinearModel/LinearModelAnalysis.cxx]: Ok. Fixed at 335280ac5

**Review by jschueller — APPROVED** [review]

**Review by sofianehaddad — APPROVED** [review]

File: lib/src/Uncertainty/Algorithm/MetaModel/LinearModel/LinearModelAnalysis.cxx [hunk 2]
```diff
@@ -432,7 +430,7 @@ Graph LinearModelAnalysis::drawResidualsVsFitted() const
...
```
Comment (josephmure) [inline lib/src/Uncertainty/Algorithm/MetaModel/LinearModel/LinearModelAnalysis.cxx:433]: Why don't you take the minimum between size and this ResourceMap key, as in drawModelVsFitted?

File: lib/src/Uncertainty/Algorithm/MetaModel/LinearModel/LinearModelAnalysis.cxx [hunk 3]
```diff
@@ -480,7 +478,7 @@ Graph LinearModelAnalysis::drawScaleLocation() const
   Cloud cloud(dataFull, "black", "fcircle");
   graph.add(cloud);
   // Add point identifiers for worst standardized residuals
-  UnsignedInteger identifiers(ResourceMap::GetAsUnsignedInteger("LinearModelAnalysis-Identifiers"));
+  UnsignedInteger identifiers(ResourceMap::GetAsUnsignedInteger("LinearModelAnalysis-IdentifiersNumber"));
```
Comment (josephmure) [inline lib/src/Uncertainty/Algorithm/MetaModel/LinearModel/LinearModelAnalysis.cxx:481]: Same (here and elsewhere in this file).

File: python/doc/examples/meta_modeling/kriging_metamodel/plot_kriging_advanced.py [hunk 4]
```diff
@@ -228,8 +228,9 @@
...
 
 # %%
-validation = ot.MetaModelValidation(x_valid, y_valid, krigingMeta)
-validation.computePredictivityFactor()
+metamodelPredictions = krigingMeta(x_valid)
+validation = ot.MetaModelValidation(y_valid, metamodelPredictions)
+validation.computeR2Score()
 
 # %%
 graph = validation.drawValidation()
```
Comment (josephmure) [inline python/doc/examples/meta_modeling/kriging_metamodel/plot_kriging_advanced.py:236]: Can you please add the following correction to this file ? Replace the line

# We have to run the opitmization algorithm again.

with

# We have to run the optimization algorithm again.

File: lib/src/Uncertainty/Algorithm/MetaModel/LinearModel/LinearModelAnalysis.cxx (hunk 2 above)
Comment (mbaudin47) [reply lib/src/Uncertainty/Algorithm/MetaModel/LinearModel/LinearModelAnalysis.cxx:433]: That is a bug: good catch!

File: lib/src/Uncertainty/Algorithm/MetaModel/LinearModel/LinearModelAnalysis.cxx (hunk 3 above)
Comment (mbaudin47) [reply lib/src/Uncertainty/Algorithm/MetaModel/LinearModel/LinearModelAnalysis.cxx]: Fixed at 8efd36c846ba2aa260d71793a6c3c0e60986a8fc.

File: python/doc/examples/meta_modeling/kriging_metamodel/plot_kriging_advanced.py (hunk 4 above)
Comment (mbaudin47) [reply python/doc/examples/meta_modeling/kriging_metamodel/plot_kriging_advanced.py:236]: Ok. Fixed at 197b25abb.

**Comment by mbaudin47** [comment]
@regislebrun : Can you please have a look at 57b1c4e78695fedf2b19bd6154ddb1ca8113e899, and see if that corresponds to what you asked for?

**Comment by mbaudin47** [comment]
@regislebrun : Can you please have a look at 57b1c4e78695fedf2b19bd6154ddb1ca8113e899, and see if that corresponds to what you asked for?

**Comment by mbaudin47** [comment]
@regislebrun , @jschueller , @josephmure : I am waiting for your reviews since 2 weeks now. Let me explain why: this PR is required by #2330, so that we can finally access to the cross-validation of a PCE. Can you please share your review on this PR?

**Review by josephmure — APPROVED** [review]

File: lib/src/Uncertainty/Algorithm/MetaModel/LinearModel/LinearModelAnalysis.cxx [hunk 5]
```diff
@@ -432,7 +430,7 @@ Graph LinearModelAnalysis::drawResidualsVsFitted() const
   Cloud cloud(dataFull, "black", "fcircle");
   graph.add(cloud);
   // Add point identifiers for worst residuals
-  UnsignedInteger identifiers(ResourceMap::GetAsUnsignedInteger("LinearModelAnalysis-Identifiers"));
+  UnsignedInteger identifiers(std::min(size, ResourceMap::GetAsUnsignedInteger("LinearModelAnalysis-IdentifiersNumber")));
```
Comment (jschueller) [inline lib/src/Uncertainty/Algorithm/MetaModel/LinearModel/LinearModelAnalysis.cxx:433]: const

File: lib/src/Uncertainty/Algorithm/MetaModel/LinearModel/LinearModelAnalysis.cxx [hunk 6]
```diff
@@ -480,7 +478,7 @@ Graph LinearModelAnalysis::drawScaleLocation() const
   Cloud cloud(dataFull, "black", "fcircle");
   graph.add(cloud);
   // Add point identifiers for worst standardized residuals
-  UnsignedInteger identifiers(ResourceMap::GetAsUnsignedInteger("LinearModelAnalysis-Identifiers"));
+  UnsignedInteger identifiers(std::min(size, ResourceMap::GetAsUnsignedInteger("LinearModelAnalysis-IdentifiersNumber")));
```
Comment (jschueller) [inline lib/src/Uncertainty/Algorithm/MetaModel/LinearModel/LinearModelAnalysis.cxx:481]: const

File: lib/src/Uncertainty/Algorithm/MetaModel/LinearModel/LinearModelAnalysis.cxx [hunk 7]
```diff
@@ -521,7 +519,7 @@ Graph LinearModelAnalysis::drawQQplot() const
   graph.setYTitle("Theoretical Quantiles");
 
   // Add point identifiers for worst standardized residuals
-  UnsignedInteger identifiers(ResourceMap::GetAsUnsignedInteger("LinearModelAnalysis-Identifiers"));
+  UnsignedInteger identifiers(ResourceMap::GetAsUnsignedInteger("LinearModelAnalysis-IdentifiersNumber"));
```
Comment (jschueller) [inline lib/src/Uncertainty/Algorithm/MetaModel/LinearModel/LinearModelAnalysis.cxx:522]: const

File: lib/src/Uncertainty/Algorithm/MetaModel/LinearModel/LinearModelAnalysis.cxx [hunk 8]
```diff
@@ -565,7 +563,7 @@ Graph LinearModelAnalysis::drawCookDistance() const
   const Point cookdistances(linearModelResult_.getCookDistances());
   const UnsignedInteger size(cookdistances.getSize());
   // Add point identifiers for worst Cook's distance
-  UnsignedInteger identifiers(ResourceMap::GetAsUnsignedInteger("LinearModelAnalysis-Identifiers"));
+  UnsignedInteger identifiers(std::min(size, ResourceMap::GetAsUnsignedInteger("LinearModelAnalysis-IdentifiersNumber")));
```
Comment (jschueller) [inline lib/src/Uncertainty/Algorithm/MetaModel/LinearModel/LinearModelAnalysis.cxx:566]: const

File: lib/src/Uncertainty/Algorithm/MetaModel/LinearModel/LinearModelAnalysis.cxx [hunk 9]
```diff
@@ -624,7 +622,7 @@ Graph LinearModelAnalysis::drawResidualsVsLeverages() const
   Cloud cloud(dataFull, "black", "fcircle");
   graph.add(cloud);
   // Add point identifiers for worst Cook's distance
-  UnsignedInteger identifiers(ResourceMap::GetAsUnsignedInteger("LinearModelAnalysis-Identifiers"));
+  UnsignedInteger identifiers(std::min(size, ResourceMap::GetAsUnsignedInteger("LinearModelAnalysis-IdentifiersNumber")));
```
Comment (jschueller) [inline lib/src/Uncertainty/Algorithm/MetaModel/LinearModel/LinearModelAnalysis.cxx:625]: const

File: lib/src/Uncertainty/Algorithm/MetaModel/LinearModel/LinearModelAnalysis.cxx [hunk 10]
```diff
@@ -716,7 +714,7 @@ Graph LinearModelAnalysis::drawCookVsLeverages() const
...
```
Comment (jschueller) [inline lib/src/Uncertainty/Algorithm/MetaModel/LinearModel/LinearModelAnalysis.cxx:717]: const

**Comment by mbaudin47** [comment]
@regislebrun Can you review this PR please?

File: lib/src/Uncertainty/Algorithm/MetaModel/LinearModel/LinearModelAnalysis.cxx (hunk 7 above)
Comment (mbaudin47) [reply lib/src/Uncertainty/Algorithm/MetaModel/LinearModel/LinearModelAnalysis.cxx:518]: Ok, Fixed.

File: lib/src/Uncertainty/Algorithm/MetaModel/LinearModel/LinearModelAnalysis.cxx (hunk 10 above)
Comment (mbaudin47) [reply lib/src/Uncertainty/Algorithm/MetaModel/LinearModel/LinearModelAnalysis.cxx]: Ok, Fixed.

File: lib/src/Uncertainty/Algorithm/MetaModel/LinearModel/LinearModelAnalysis.cxx (hunk 9 above)
Comment (mbaudin47) [reply lib/src/Uncertainty/Algorithm/MetaModel/LinearModel/LinearModelAnalysis.cxx]: Ok, Fixed.

File: lib/src/Uncertainty/Algorithm/MetaModel/LinearModel/LinearModelAnalysis.cxx (hunk 8 above)
Comment (mbaudin47) [reply lib/src/Uncertainty/Algorithm/MetaModel/LinearModel/LinearModelAnalysis.cxx]: Ok, Fixed.

File: lib/src/Uncertainty/Algorithm/MetaModel/LinearModel/LinearModelAnalysis.cxx (hunk 6 above)
Comment (mbaudin47) [reply lib/src/Uncertainty/Algorithm/MetaModel/LinearModel/LinearModelAnalysis.cxx]: Ok, Fixed.

File: lib/src/Uncertainty/Algorithm/MetaModel/LinearModel/LinearModelAnalysis.cxx (hunk 5 above)
Comment (mbaudin47) [reply lib/src/Uncertainty/Algorithm/MetaModel/LinearModel/LinearModelAnalysis.cxx]: Ok, Fixed.

**Comment by mbaudin47** [comment]
@jschueller : Given the lack of review of @regislebrun and the 3 approvals, I suggest to merge this PR.

**Pull request merged**